### ISSUE TITLE
Random non-idempotent ansible tasks

### ISSUE THREAD

**Issue by rattboi** [BonnyCI/projman#38]
As we are right now, every run of system-ansible has 14 changes at stable state, and cideploy has 5 (3 on one host, 2 on another). It'd be nice to get that down to zero.

I enabled even more verbose logs for a minute, and here are links to those, which should help figure out before/after things.

System Ansible: http://contrasjc-bastion.portbleu.com/cron-logs/system-ansible/ansible_bastion.yml_20161203031908.log

cideploy: http://contrasjc-bastion.portbleu.com/cron-logs/cideploy/ansible_install-ci.yml_20161203033129.log

**Comment by missaugustina** [comment]
What's the status of this issue?

**Comment by rattboi** [comment]
Looks like this is getting better. 

cideploy has 1 non-idempotent task, and system-ansible has 11. Getting the system-ansible one down would be great, but I think it's happening because of a larger issue. 

I believe there's some overlapping tasks being run by root, then cideploy, every system-ansible run. This causes each set to have changes every time.

I'll look into it further.

**Comment by mlangbehn** [comment]
This is required for BonnyCI/projman#49.

**Comment by missaugustina** [comment]
This isn't really required for that ticket but it is dependent on it. The check on the tests need to be aware of what's changed. If those tests exist then this work would need to update those tests.

**Comment by mlangbehn** [comment]
**Summary of Issues:**

* ansible-runner role modifies file permissions depending on which user is running the role
  * BonnyCI/hoist#182 resolves these issues
* [tailon : enable tailon service] also shows changed on every run [**EDIT:** This seems to only happen in Vagrant]
  * Working on this one still
    * Seems that the systemd module is reporting changed every time we call it to enable and start the tailon service
    * For kicks and laughs I tested this with the service module and the issue persisted
* Secrets file is changing permissions
  * Opened BonnyCI/projman#154 to track this one

**Comment by missaugustina** [comment]
Do we know what additional items need to be done on this? If not, what are the next steps to finding out?

**Comment by mlangbehn** [comment]
This is still in progress. It was on hold while I was working on getting vagrant/docker deploys working to continue testing.

Will update after testing.

**Comment by mlangbehn** [comment]
~Blocked by~:
* ~BonnyCI/hoist#242~
* ~BonnyCI/hoist#243~

**Comment by mlangbehn** [comment]
Testing via `tools/vagrant-deploy.sh` and `tools/docker-deploy.sh` shows that the only item showing as changed on a second run is the `apache restart` handler. This handler is being run twice on the elk node, and twice on the logs node (or a total of four times on the docker `allinone` node).

**Comment by mlangbehn** [comment]
I'm not making much progress with the apache role for the following reasons:

* No obvious issue in the code is standing out to me
* Running only the apache role does not reproduce the issue
* Running only the `zuul` or `logs` tags fails for an unrelated reason
  * Running with `--tags zuul,logs` also fails
